File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

class Solution {
    public ListNode reverseList(ListNode head) {

        // Create Stack 
        Stack<ListNode> stack = new Stack<ListNode>();

        while(head!=null) {
            stack.push(head);
            head  = head.next;
        }

        ListNode dummy = new ListNode(-1);
        head = dummy;
        while(!stack.isEmpty()) {
            ListNode node = stack.pop();
            head.next = new ListNode(node.val);
            head = head.next;
        }

        return dummy.next;
    }
}

File: Data Structures & Algorithms/palindrome-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def isPalindrome(self, head: Optional[ListNode]) -> bool:
        """ 

            create a list - push the elements to the list

            do two pointer techique on the list
        """

        res = list()
        curr = head
        if not curr:
            return res
        while curr:
            res.append(curr.val)
            curr = curr.next
        

        # palindrme check 
        start = 0
        end = len(res)-1
        while start < end:
            if res[start] == res[end]:
                start += 1
                end -= 1
            else:
                return False

        return True
